hamilton/reframe_parameterisation_infrastructure/reframe_tests/parameterisation_tests.py

```diff
@@ -71,10 +72,19 @@ def __init__(self, n_processors, test_id, iteration):
         self.prerun_cmds = [
             f"pushd {execution_dir}",
             f"rm -rf pupil_data_output_{test_id}_{iteration}.csv",  # prevent appending to previous data
+            "echo $PATH",
+            "source $HOME/.bashrc",
+            "conda init",
             "source activate classroom_abm",
         ]
 
-        self.executable = "python run_pipeline.py"
+        if os.environ.get("HAMILTON_VERSION") == "hamilton8":
+            # Need to manually load modules on hamilton8 as modules system isn't compatible with ReFrame
+            self.prerun_cmds.extend(
+                ["module load r/4.1.2", "module load $R_BUILD_MODULES"],
```

Comment:
Was it necessary to take this out of the system config level? I saw in this branch's system config for Hamilton 8 `module_system` is set to `nomod` - I'm guessing you tried the other options that ReFrame supports here and they didn't work?

Reply:
The module system on Hamilton 8 is a custom one, so doesn't work in the way reframe expects, so none of its options for module systems worked! So I had to move the module loading out of reframe unfortunately.

Reply:
Ah that's a shame - I've just seen the path to the modules bin: `/apps/infrastructure/modules/default/default/default/Modules/default/bin/`. Do you know if this is how Hamilton 8 will be configured in the long term or if it is just a temporary placeholder?

Reply:
I'm not sure. I let Karen know about reframe though.